### ISSUE TITLE
request: add broadPeaks and gappedPeaks format options

### ISSUE THREAD

**Issue by AdrijaK** [deeptools/pyGenomeTracks#206]
It would be nice if pyGenomeTracks would recognize broadpeak and narrowPeak tracks as well. 

### broadPeak
The broadPeak file:
```bash
# head -n 2 H3K4me1-dox-rep1.filtered.BAM_peaks.broadPeak
JH584304.1	26816	27277	H3K4me1-dox-rep1.filtered.BAM_peak_1	53	.	1.47245	6.96058	5.38358
JH584304.1	38249	38775	H3K4me1-dox-rep1.filtered.BAM_peak_2	25	.	1.36614	3.93211	2.54933
```

The plotting command:
```bash
pyGenomeTracks --tracks tmp.tracks.ini --region chr11:18850000-19050000 --title "example, zoomed" --outFileName tmp.example.pdf
```

the `tmp.tracks.ini` with `file_type = narrow_peak`:
```
[macs2 broadPeak]
file = H3K4me1-dox-rep1.filtered.BAM_peaks.broadPeak
file_type = narrow_peak
```
error:
```
[macs2 broadPeak]
Traceback (most recent call last):
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/bin/pyGenomeTracks", line 11, in <module>
    main(args)
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/plotTracks.py", line 312, in main
    h_align_titles=args.trackLabelHAlign)
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/tracksClass.py", line 270, in plot
    track.plot(plot_axis, chrom, start, end)
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/tracks/NarrowPeakTrack.py", line 111, in plot
    name, score, strand, signal_value, p_value, q_value, summit = peak
ValueError: not enough values to unpack (expected 7, got 6)
```

the `tmp.tracks.ini` with `file_type = bed`:
```
[macs2 broadPeak]
file = H3K4me1-dox-rep1.filtered.BAM_peaks.broadPeak
file_type = bed
```
error message:
```
[macs2 broadPeak]
  0%|                                                | 0/114530 [00:00<?, ?it/s]Value: 1.47245 in field 7 at line 2 is not an integer
Traceback (most recent call last):
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/bin/pyGenomeTracks", line 11, in <module>
    main(args)
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/plotTracks.py", line 279, in main
    trp = PlotTracks(args.tracks.name, args.width, fig_height=args.height, fontsize=args.fontSize, dpi=args.dpi, track_label_width=args.trackLabelFraction)
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/tracksClass.py", line 111, in __init__
    self.track_obj_list.append(track_class(properties))
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/tracks/BedTrack.py", line 171, in __init__
    self.interval_tree, min_score, max_score = self.process_bed()
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/tracks/BedTrack.py", line 231, in process_bed
    for bed in tqdm(bed_file_h, total=total_length):
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/tqdm/std.py", line 1119, in __iter__
    for obj in iterable:
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/readBed.py", line 140, in __next__
    if self.prev_chrom == bed.chromosome:
AttributeError: 'dict' object has no attribute 'chromosome'
```

### gappedPeak
```bash
# head -n 2 H3K4me1-dox-rep1.filtered.BAM_peaks.gappedPeak
JH584304.1	26816	27277	H3K4me1-dox-rep1.filtered.BAM_peak_1	53	.	26816	27277	0	3	1,409,1	0,35,460	1.47245	6.96058	5.38358
JH584304.1	38249	38775	H3K4me1-dox-rep1.filtered.BAM_peak_2	25	.	38249	38775	0	2	1,1	0,525	1.36614	3.93211	2.54933
```

the `tmp.tracks.ini` with `file_type = narrow_peak`:
```
[macs2 gappedPeak]
file = H3K4me1-dox-rep1.filtered.BAM_peaks.gappedPeak
file_type = narrow_peak
```
error message:
```
[macs2 gappedPeak]
Traceback (most recent call last):
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/bin/pyGenomeTracks", line 11, in <module>
    main(args)
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/plotTracks.py", line 312, in main
    h_align_titles=args.trackLabelHAlign)
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/tracksClass.py", line 270, in plot
    track.plot(plot_axis, chrom, start, end)
  File "/localhome/bric/qlr900/analysis/software/anaconda3/envs/pygenometracks/lib/python3.6/site-packages/pygenometracks/tracks/NarrowPeakTrack.py", line 111, in plot
    name, score, strand, signal_value, p_value, q_value, summit = peak
ValueError: too many values to unpack (expected 7)
```

**Comment by lldelisle** [comment]
Hi,
Thank you very much for reporting this error.
I thought file_type = bed would work but it did not. I am working on a fix.
Which type of plotting would you like?
Something like the narrowPeak without summit with the possibility to plot as bells, or as box with
 score, p-value etc. or just like bed?
![master_narrowPeak2](https://user-images.githubusercontent.com/30404086/78913081-c22b9b00-7a88-11ea-8ab6-3ec740241680.png)

**Comment by AdrijaK** [comment]
Hi Lucille, thank you for the very fast reply. Both "score, p-value etc" or "bed" options are great options. 

For context, I try to use pyBedTools to evaluate the results of snakePipes pipelines, and my main aim is to check how different peak caller outputs agree with each other. They use macs2, genrich, histoneHMM for ChIP peaks. Since this pipeline outputs bed files in slightly different flavors (see https://github.com/maxplanck-ie/snakepipes/issues/611 and https://github.com/maxplanck-ie/snakepipes/issues/613 ) it would also make a lot of sense if I could specify whether the given file is "bed3", "bed4", "bed5" or "bed6" (the default). I use [bedtools definitions](https://bedtools.readthedocs.io/en/latest/content/general-usage.html) to refer to bed formats. 

If broadPeaks and gappedPeaks work as their own format, or alternatively as "bed", that would be brilliant.

**Comment by lldelisle** [comment]
Hi,
On my current bug fix, the `file_type = bed` will adapt to the bed and guess the bed3, 4, 5... which fits with the first line.
For the moment, it is working with broadPeak, gappedPeak and the strange format of Filtered results.
```
chr1	6405901	6407100	1200	*	65.683696844062	8	8	0	3.33261803969157e-08	2.70165765351933e-07	up	1.37928104261085	6406501	region10
chr1	87947251	87947700	450	*	31.457878109242	3	0	3	0.000216645911052967	0.000714845502415761	down	-1.24101800381362	87947551region1000

```
Could you give me an example of the histoneHMM output?

**Comment by lldelisle** [comment]
For the moment, this is where I am:
```
[macs2 broadPeak]
file = broadPeak.broadPeak
title = broadPeak
file_type = bed

[spacer]

[macs2 gappedPeak]
file = gappedPeak.gappedPeak
title = gappedPeak
file_type = bed

[spacer]

[macs2 filteredbed]
file = strange.filtered.bed
title = filtered.bed (strange format)
file_type = bed
```
![master_broadPeak](https://user-images.githubusercontent.com/30404086/78941096-4ac13000-7ab7-11ea-95cc-49b1424f9b06.png)

**Comment by AdrijaK** [comment]
Thank you. Encountering "strange" formats (such as  filtered.bed) is the main reason I think allowing the user to specify "BED3" vs"BED6" would still be a good idea: then you could make pyGenomeTracks read only the first three columns of the "strange" files as bed3 (ignoring the rest of "strange" things within the file). What do you think?

**Comment by lldelisle** [comment]
Hi,
We just released version 3.4 which will guess the bed type. If you still find examples where you need to specify BED3 or BED6, feel free to reopen the issue and I will implement this option.
